my_enumerable.rb

```diff
@@ -0,0 +1,185 @@
+    end
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    idx = 0
+    while idx < to_a.length
+      yield(to_a[idx], idx)
+      idx += 1
+    end
+    self
+  end
+
+  def my_select
+    return to_enum unless block_given?
+
+    select_arr = []
+    my_each { |value| select_arr << value if yield(value) }
+    select_arr
+  end
+
+  def my_all?(arg = nil)
+    if block_given?
+      to_a.my_each { |value| return false if yield(value) == false }
+      return true
+    elsif arg.nil?
+      to_a.my_each { |value| return false if value.nil? || value == false }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |value| return false unless [value.class, value.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |value| return false unless arg.match(value) }
+    else
+      to_a.my_each { |value| return false if value != arg }
+    end
+    true
+  end
+
+  def my_any?(arg = nil)
+    if block_given?
+      to_a.my_each { |idx| return true if yield(idx) }
+      return false
+    elsif arg.nil?
+      to_a.my_each { |idx| return true if idx }
+    elsif !arg.nil? && (arg.is_a? Class)
+      to_a.my_each { |idx| return true if [idx.class, idx.class.superclass].include?(arg) }
+    elsif !arg.nil? && arg.class == Regexp
+      to_a.my_each { |idx| return true if arg.match(idx) }
+    else
+      to_a.my_each { |idx| return true if idx == arg }
+    end
+    false
+  end
+
+  def my_none?(arg = nil, &block)
+    !my_any?(arg, &block)
+  end
+
+  def my_count(arg = nil)
+    count = 0
+    if arg.nil? && block_given?
+      my_each { |value| count += 1 if yield(value) }
+    elsif arg.nil?
+      my_each { |_value| count += 1 }
+    else
+      my_each { |value| count += 1 if value == arg }
+    end
+  end
+
+  def my_map(proc = nil)
+    return to_enum unless block_given?
+
+    result = []
+    proc.nil? ? my_each { |idx| result << yield(idx) } : my_each { |idx| result << proc.call(idx) }
+    result
+  end
+
+  def my_inject(arg1 = nil, arg2 = nil)
+    array = is_a?(Array) ? self : to_a
+    symbol_ = arg1 if arg1.is_a?(Symbol) || arg1.is_a?(String)
+    actual_item = arg1 if arg1.is_a? Integer
+
+    if arg1.is_a?(Integer)
+      if arg2.is_a?(Symbol) || arg2.is_a?(String)
+        symbol_ = arg2
+      elsif !block_given?
+        raise "#{arg2} is not a symbol nor a string"
+      end
+    elsif arg1.is_a?(Symbol) || arg1.is_a?(String)
+      raise "#{arg2} is not a symbol nor a string" if !arg2.is_a?(Symbol) && !arg2.nil?
+
+      raise "undefined method `#{arg2}' for :#{arg2}:Symbol" if arg2.is_a?(Symbol) && !arg2.nil?
+    end
+
+    if symbol_
+      array.my_each { |choice| actual_item = actual_item ? actual_item.send(symbol_, choice) : choice }
+    elsif block_given?
+      array.my_each { |choice| actual_item = actual_item ? yield(actual_item, choice) : choice }
+    else
+      raise 'no block given'
+    end
+    actual_item
+  end
```

Comment:
- [ ] should have working `my_inject` identical and returns the same thing as ruby's [`inject`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-inject). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_inject.png) from [Odin](https://github.com/TheOdinProject/curriculum/blob/master/ruby_programming/archive/basic_ruby/project_advanced_building_blocks.md#project-2-enumerable-methods))

####  my_inject
> Remember to make it work for ranges and hashes
- [ ] combines all elements of enum by applying a binary operation, specified by a block
```ruby
(1..5).my_inject(4) { |prod, n| prod * n } == (1..5).inject(4) { |prod, n| prod * n } => 480
```
- my_inject is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95801703-73b9bc00-0cb8-11eb-89f4-15d632186200.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698409-2ffaa180-6f76-11ea-91d3-9778aa560411.png)

- [ ] When a symbol is provided as an argument, my_inject should combine the elements of the enum using that symbol as a named method.
```ruby
[2, 5, 6].my_inject(:+) == [2, 5, 6].inject(:+) => 13
```
- Yours is retrieving an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95801778-ae235900-0cb8-11eb-95a4-5aa17ee76d5b.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/77698534-6a643e80-6f76-11ea-9d0e-ebb0f02da179.png)

- [ ] when a symbol is specified with an initail value combines each element of the collection by applying the symbol as a named method
```ruby
[2, 5, 6].my_inject(20, :+) == [2, 5, 6].inject(20, :+) => 33
```
- my_inject retrieves an ArgumentError
![image](https://user-images.githubusercontent.com/52209067/95801877-02c6d400-0cb9-11eb-8f08-3625f22852db.png)

- Real inject
![image](https://user-images.githubusercontent.com/52209067/95801897-18d49480-0cb9-11eb-8495-7fbc48680f78.png)